assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js

```diff
@@ -99,10 +100,44 @@ export default function UserDimensionsPieChart( {
 			}
 		};
 
+		const closeToolTip = () =>
+			setValues( {
+				[ UI_DIMENSION_VALUE ]: '',
+				[ UI_DIMENSION_COLOR ]: '',
+				[ UI_ACTIVE_ROW_INDEX ]: null,
+			} );
+
+		const tooltipIsOpen =
+			!! UI_ACTIVE_ROW_INDEX &&
+			!! UI_DIMENSION_COLOR &&
+			!! UI_DIMENSION_VALUE;
```

Comment:
This is always `true` because you check non-empty constants. I think the most reliable way to check whether the tooltip is open is to check whether the appropriate element exists on the page using vanilla js. We can do it something like this:

```javascript
const onEscape = ( event = {} ) => {
    if ( event?.keyCode === ESCAPE && !! currentContainerRef.querySelector( '.googlesitekit-visualization-tooltip' ) ) {
        closeToolTip();
    }
};

```

Reply:
Good idea thanks. Not sure why I was using the constants there, that's ridiculous 😂

Reply:
Also: I don't know why the linting issue wasn't picked up, but I've addressed it.